File: sn/src/main/java/org/sqlbuilder/sn/SnModule.java
package org.sqlbuilder.sn;

import org.sqlbuilder.common.Module;

import java.io.IOException;

public class SnModule extends Module
{
	public static final String MODULE_ID = "sn";

	protected SnModule(final String conf, final Mode mode)
	{
		super(MODULE_ID, conf, mode);
	}

	@Override
	protected void run()
	{
		assert props != null;

		try
		{
			switch (mode)
			{
				case PLAIN:
					new SnProcessor(props).run();
					break;

				case RESOLVE:
					new SnResolvingProcessor(props).run();
					break;

				case UPDATE:
					new SnUpdatingProcessor(props).run();
					break;

				case EXPORT:
				default:
					return;
			}
		}
		catch (IOException | ClassNotFoundException e)
		{
			e.printStackTrace();
		}
	}

	public static void main(final String[] args) throws IOException
	{
		int i = 0;
		Mode mode = Mode.PLAIN;
		if (args[i].startsWith("-"))
		{
			mode = Mode.read(args[i++]);
		}
		String conf = args[i];
		new SnModule(conf, mode).run();
	}
}

File: sn/src/main/java/org/sqlbuilder/sn/SnModule.kt
package org.sqlbuilder.sn

import org.sqlbuilder.common.Module
import org.sqlbuilder.common.Module.Mode.Companion.read
import java.io.IOException

class SnModule(
    conf: String, mode: Mode,
) : Module(MODULE_ID, conf, mode) {

    override fun run() {
        checkNotNull(props)

        try {
            when (mode) {
                Mode.PLAIN   -> SnProcessor(props).run()
                Mode.RESOLVE -> SnResolvingProcessor(props).run()
                Mode.UPDATE  -> SnUpdatingProcessor(props).run()
                Mode.EXPORT  -> {}
                else         -> {}
            }
        } catch (e: IOException) {
            e.printStackTrace()
        } catch (e: ClassNotFoundException) {
            e.printStackTrace()
        }
    }

    companion object {

        const val MODULE_ID: String = "sn"

        fun main(args: Array<String>) {
            var i = 0
            var mode = Mode.PLAIN
            if (args[i].startsWith("-")) {
                mode = read(args[i++])
            }
            val conf = args[i]
            SnModule(conf, mode).run()
        }
    }
}
